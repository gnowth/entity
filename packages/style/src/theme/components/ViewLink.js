import { css } from 'styled-components';

import { color } from '../../selectors';

// eslint-disable-next-line import/prefer-default-export
export const component_viewLink_standard = css`
  color: ${color({ name: 'black' })};
`;
