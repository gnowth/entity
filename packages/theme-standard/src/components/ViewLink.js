import { color } from '@gnowth/theme';
import { css } from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const component_viewLink_standard = {
  css: css`
    color: ${color({ palette: 'black' })};
  `,
};
