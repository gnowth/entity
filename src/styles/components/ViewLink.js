import { color } from '@gnowth/style';
import { css } from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const main = css`
  color: ${color({ name: 'black' })};
`;
