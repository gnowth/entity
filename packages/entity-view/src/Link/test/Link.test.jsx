import renderer from 'react-test-renderer';
import React from 'react';
import { EntityTitle, Fields } from '@entity/core';
import { HashRouter as Router } from 'react-router-dom';

import ViewLink from '../Link';

class EntityTest extends EntityTitle {
  static paths = {
    urlBase: '/test/',
  }
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Router>
      <ViewLink
        field={new Fields.EntityField({ entity: EntityTest })}
        value={EntityTest.dataToRecord({})}
      />
    </Router>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
