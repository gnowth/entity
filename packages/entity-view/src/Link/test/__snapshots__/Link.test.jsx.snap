// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="Link-sc-1hs1rmw-0 jfsnLe"
  href="#/test/undefined/"
  onClick={[Function]}
  value={
    Immutable.Map {
      "description": "",
      "is_archived": false,
      "locale": null,
      "order": null,
      "title": "",
      "title_short": "",
      "uuid": undefined,
    }
  }
>
  View
</a>
`;
